src/components/TrackTile/TrackerDetails/TrackerHistoryChart/TrackerHistoryChart.tsx

```diff
@@ -23,37 +23,42 @@ type TrackerHistoryChartProps = {
   unit?: string;
   stepperPosition?: 'top' | 'bottom';
   dateRangeType?: DateRangeType;
+  referenceDate?: Date;
 } & Pick<ChartProps, 'variant' | 'color'>;
 
-const getInitialRange = (dateRangeType: DateRangeType) => {
+const getInitialRange = (dateRangeType: DateRangeType, referenceDate: Date) => {
   switch (dateRangeType) {
     case 'calendarWeek':
       return {
-        start: startOfWeek(startOfToday(), { weekStartsOn: 1 }),
-        end: endOfWeek(startOfToday(), { weekStartsOn: 1 }),
+        start: startOfWeek(referenceDate, { weekStartsOn: 1 }),
+        end: endOfWeek(referenceDate, { weekStartsOn: 1 }),
       };
     case 'past7Days':
       return {
-        start: addDays(startOfToday(), -6),
-        end: endOfToday(),
+        start: addDays(referenceDate, -6),
+        end: endOfDay(referenceDate),
       };
   }
 };
 
 export const TrackerHistoryChart: FC<TrackerHistoryChartProps> = (props) => {
   const theme = useTheme();
   const { metricId, stepperPosition = 'top', tracker, valuesContext } = props;
-  const { variant, target, unit, dateRangeType = 'past7Days' } = props;
-  const [dateRange, setDateRange] = useState(getInitialRange(dateRangeType));
+  const {
+    variant,
+    target,
+    unit,
+    dateRangeType = 'past7Days',
+    referenceDate = startOfToday(),
+  } = props;
+  const [dateRange, setDateRange] = useState(
+    getInitialRange(dateRangeType, referenceDate),
+  );
   const { loading, error, trackerValues } = useTrackerValues(
     valuesContext,
     dateRange,
   );
 
-  useEffect(() => {
-    setDateRange(getInitialRange(dateRangeType));
-  }, [dateRangeType]);
-
```

Comment:
This is sort of unrelated but noticed the potential here for unnecessary re-renders. I wouldn't expect this to be a prop that would change (basically just toggles between a calendar week and last 7 days) but it would fall under [these conditions](https://react.dev/learn/you-might-not-need-an-effect#resetting-all-state-when-a-prop-changes) where `useEffect` isn't the best choice.